core/src/main/java/org/jdbi/v3/core/statement/TemplateEngine.java

```diff
@@ -37,4 +43,29 @@
      * and which can bind the correct arguments to that prepared statement
      */
     String render(String template, StatementContext ctx);
+
+    /**
+     * Parse a SQL template and return a parsed representation ready to apply to a statement.
+     * @param template the sql template to parse
+     * @param config the Jdbi configuration at prepare time
+     * @return a parsed representation, if available
+     */
+    @Beta
+    default Optional<Function<StatementContext, String>> parse(String template, ConfigRegistry config) {
+        return Optional.empty();
+    }
+
+    @FunctionalInterface
+    @Beta
+    interface Caching extends TemplateEngine {
```

Comment:
This feels like a misnomer. There's no caching happening here, even though the parse method is making caching possible elsewhere.